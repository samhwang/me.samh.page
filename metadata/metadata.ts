// Icons List. See https://devicon.dev/
const ICONS = {
  PHP: 'php-plain',
  CODEIGNITER: 'codeigniter-plain',
  LARAVEL: 'laravel-plain',
  COMPOSER: 'composer-plain',
  DRUPAL: 'drupal-plain',
  WORDPRESS: 'wordpress-plain',
  WOOCOMMERCE: 'woocommerce-plain',
  JS: 'javascript-plain',
  TS: 'typescript-plain',
  JQUERY: 'jquery-plain',
  D3JS: 'd3js-plain',
  REACT: 'react-original',
  REDUX: 'redux-original',
  GATSBY: 'gatsby-plain',
  ANGULAR: 'angularjs-plain',
  MATERIALUI: 'materialui-plain',
  STORYBOOK: 'storybook-plain',
  NODE: 'nodejs-plain',
  EXPRESS: 'express-original',
  METEOR: 'meteor-plain',
  GRAPHQL: 'graphql-plain',
  SOCKETIO: 'socketio-original',
  JEST: 'jest-plain',
  MOCHA: 'mocha-plain',
  CUCUMBER: 'cucumber-plain',
  BABEL: 'babel-plain',
  WEBPACK: 'webpack-plain',
  NPM: 'npm-original-wordmark',
  YARN: 'yarn-plain',
  GO: 'go-original-wordmark',
  HTML5: 'html5-plain',
  CSS3: 'css3-plain',
  SASS: 'sass-orignal',
  LESS: 'less-plain-wordmark',
  BOOTSTRAP: 'bootstrap-plain',
  MONGODB: 'mongodb-plain',
  MYSQL: 'mysql-plain',
  POSTGRES: 'postgresql-plain',
  AWS: 'amazonwebservices-original',
  AZURE: 'azure-plain',
  DIGITALOCEAN: 'digitalocean-plain',
  DOCKER: 'docker-plain',
  GIT: 'git-plain',
  GITHUB: 'github-original',
  GITLAB: 'gitlab-plain',
  LINKEDIN: 'linkedin-plain',
  JIRA: 'jira-plain',
  BITBUCKET: 'bitbucket-plain',
  CONFLUENCE: 'confluence-original',
  TRELLO: 'trello-plain',
  ESLINT: 'eslint-original',
};

// Manifest for PWA.
export const manifest = {
  name: 'Sam Huynh Website',
  short_name: 'Sam Huynh',
  start_url: '/',
  background_color: '#FFFFFF',
  theme_color: '#FFFFFF',
  display: 'standalone',
  icon: 'src/assets/img/website-icon.png',
};

const metadata = {
  siteTitle: 'Sam Huynh',
  siteUrl: 'https://samhwang.github.io',
  siteDescription: 'Online profile of Sam Huynh.',
  manifest,
  personalInformation: {
    firstName: 'Sam',
    lastName: 'Huynh',
    bio: 'I have always been passionate about computers, and how technology can bring the world closer. That was the reason why I walked down the path of Software Development. Besides my studies, I have worked in a lot of Agile teams of up to 10 people, delivering high quality responsive web applications on different platforms (PHP, NodeJS) for clients in different industries (retail, sports, and education.)',
    socialLinks: [
      {
        icon: ICONS.GITHUB,
        name: 'Github',
        url: 'https://github.com/samhwang',
      },
      {
        icon: ICONS.LINKEDIN,
        name: 'Linkedin',
        url: 'https://www.linkedin.com/in/samhwang2112/',
      },
    ],
    email: 'samhwang2112.dev@gmail.com',
    address: 'Melbourne, Victoria, Australia',
    experience: [
      {
        title: 'Full Stack Software Developer',
        companyName: 'HungryHungry',
        duration: 'Jun 2021 - Present',
        description: [
          "Built and maintained various parts of, and introduce more functionality into HH's Ordering System in the course of migrating the legacy monolith tech stack into microservices: Payments, Identity, POS Integrations, Assisted Marketing, Kiosk, Bar Tabs, Notifications, ...",
          "Built and maintained HH's UI Components Library to streamline the UI Apps creation process",
          "Built and maintained HH's internal NPM Packages on CodeArtifact",
          'Automated microservices deployment using Buildkite CI/CD pipelines',
          'Assisted in interviewing and onboarding new staff members',
        ],
        techIcons: [
          ICONS.JS,
          ICONS.TS,
          ICONS.JQUERY,
          ICONS.REACT,
          ICONS.ANGULAR,
          ICONS.NODE,
          ICONS.EXPRESS,
          ICONS.NPM,
          ICONS.STORYBOOK,
          ICONS.JEST,
          ICONS.ESLINT,
          ICONS.WEBPACK,
          ICONS.PHP,
          ICONS.DRUPAL,
          ICONS.MYSQL,
          ICONS.HTML5,
          ICONS.CSS3,
          ICONS.SASS,
          ICONS.AWS,
          ICONS.DOCKER,
          ICONS.BITBUCKET,
          ICONS.JIRA,
          ICONS.CONFLUENCE,
        ],
        technologies: [
          'Front-end: React, Angular, Emotion.sh, SASS',
          'Back-end: PHP (Drupal), JavaScript/TypeScript, NodeJS (Express, Fastify, Serverless Framework)',
          'Testing: Jest, Vitest, Storybook, React Testing Library',
          'JS Code Bundler: Rollup, Webpack, ESBuild, SWC',
          "Built and maintained HH's services internal REST API",
          'Docker containers & AWS Elastic Container Services (Fargate)',
          "Integrated payments provider (Stripe, Afterpay, Braintree) into HH's system",
          "Integrated AWS Cognito into HH's Identity Service",
          'Event Driven Architecture using SQS and Kafka',
          'Database: MySQL/MariaDB, AWS RDS Aurora Serverless, DynamoDB',
          'AWS (Lambda, S3, SQS, SES, SNS, ECS, ELB, CodeArtifact)',
          'Monorepo Management with Turborepo',
          'Ran, built and maintained CI/CD pipeline with Buildkite.',
          'Error and Metrics reporting with Datadog',
        ],
      },
      {
        title: 'Full Stack Software Developer',
        companyName: 'Hertz Australia',
        duration: 'Oct 2019 - Jun 2021',
        description: [
          'Built the Github Actions and GitHub packages pipeline to publish the Flexicar website docker images, along with setting up APIs to access GitHub packages for cleaning up when necessary',
          "Built new feature into Hertz's booking API, including but not limited to integrations with insurance partners and franchisee, error monitoring, reporting rental analytics and tracking systems, using GraphQL (Apollo), MongoDB, NodeJS (Express), EJS, ReactJS, Sentry and web sockets on AWS Serverless platform, with a dashboard on Elastic Beanstalk.",
          "Rebuilt Hertz's customer portal for past journey information tand the shuttle bus tracking and booking portal, using React/Redux and Material UI",
        ],
        techIcons: [
          ICONS.JS,
          ICONS.JQUERY,
          ICONS.REACT,
          ICONS.REDUX,
          ICONS.MATERIALUI,
          ICONS.STORYBOOK,
          ICONS.NODE,
          ICONS.EXPRESS,
          ICONS.SOCKETIO,
          ICONS.GRAPHQL,
          ICONS.JEST,
          ICONS.MOCHA,
          ICONS.NPM,
          ICONS.YARN,
          ICONS.PHP,
          ICONS.WORDPRESS,
          ICONS.HTML5,
          ICONS.CSS3,
          ICONS.SASS,
          ICONS.LESS,
          ICONS.MYSQL,
          ICONS.MONGODB,
          ICONS.AWS,
          ICONS.DOCKER,
          ICONS.GITHUB,
          ICONS.JIRA,
          ICONS.CONFLUENCE,
        ],
        technologies: [
          'Front-end: React, EJS, jQuery, Bootstrap, Material UI, Jest, Storybook, sockets.io',
          'Back-end: PHP7 (Wordpress), NodeJS (Express)',
          'HTML5, CSS3 and CSS pre-processors like SASS and LESS',
          'GraphQL (Apollo) and REST API',
          'Docker containers & docker-compose',
          'Database: MySQL & MongoDB',
          'Serverless architecture and websockets',
          'AWS (Lambda, Elastic Beanstalk, SQS, SES, SNS, CloudWatch, S3)',
          'CI/CD: GitHub Actions pipeline and GitHub Container Registry',
          'Project management using JIRA and Confluence',
          'Error reporting and monitoring: Sentry, Dashbird',
        ],
      },
      {
        title: 'Web Application Developer',
        companyName: 'Bonntech Business Solutions',
        duration: 'May 2019 - Oct 2019',
        description: [
          'Built the base framework, along with its own create and update script to speed up the initial setup for future projects.',
          "Assisted in setting up Bonntech's central Git repository and private Composer Packagist for internal projects and libraries.",
        ],
        techIcons: [ICONS.PHP, ICONS.COMPOSER, ICONS.JS, ICONS.BOOTSTRAP, ICONS.CSS3, ICONS.LESS, ICONS.SASS, ICONS.MYSQL, ICONS.DOCKER, ICONS.GITLAB],
        technologies: [
          'PHP 7.3, using CodeIgniter framework and Composer package manager with self-hosted Packagist',
          'MySQL database management and data modelling',
          'HTML5, CSS, JavaScript (jQuery, Bootstrap, Datatables, and AJAX)',
          'Apache server configurations',
          'Built containerized development environments with Docker',
        ],
      },
      {
        title: 'Software Developer',
        companyName: 'Classcom',
        duration: 'Jan 2019 - Apr 2019',
        description: [
          'Collaborated on building new core features of the ClassCom project',
          'Created automated test and build scripts and participated in code reviews in a bi-weekly release schedule.',
        ],
        techIcons: [
          ICONS.JS,
          ICONS.NODE,
          ICONS.METEOR,
          ICONS.MOCHA,
          ICONS.CUCUMBER,
          ICONS.HTML5,
          ICONS.CSS3,
          ICONS.LESS,
          ICONS.MONGODB,
          ICONS.NPM,
          ICONS.AZURE,
        ],
        technologies: [
          'JavaScript, using the Meteor full stack framework (with Node/Express for backend processing, Blaze for templating, CSS pre-processing with LESS)',
          'Automated test scripting with CucumberJS and Mocha.',
          'MongoDB database and data modelling.',
          'REST APIs',
          'Project management using Azure DevOps',
        ],
      },
      {
        title: 'PHP Developer Intern',
        companyName: 'Psych Press',
        duration: 'Jun 2017 - Sep 2017',
        description: ['Collaborated on building three inhouse projects for Psych Press closely with the psychology team on a weekly basis.'],
        techIcons: [ICONS.PHP, ICONS.LARAVEL, ICONS.JS, ICONS.JQUERY, ICONS.D3JS, ICONS.BOOTSTRAP, ICONS.HTML5, ICONS.CSS3, ICONS.AWS, ICONS.BITBUCKET],
        technologies: [
          'PHP 7, using CodeIgniter, Laravel and Moodle framework.',
          'JavaScript (AJAX, jQuery, Bootstrap, D3js and ChartJS) and CSS (MaterializeCSS)',
          'Microsoft SQL Server Database',
          'Migration from Microsoft servers to Amazon Web Services EC2 Linux instances',
        ],
      },
      {
        title: 'Web Developer',
        companyName: 'Mint n Co. Accessories',
        duration: 'Oct 2015 - Feb 2016',
        description: [
          'Designed and built the e-commerce site for Mint and Co. Accessories from the ground up.',
          'Provided regular maintenance to ensure 99.9% site availability.',
        ],
        techIcons: [ICONS.PHP, ICONS.WORDPRESS, ICONS.WOOCOMMERCE, ICONS.BOOTSTRAP, ICONS.HTML5, ICONS.CSS3, ICONS.JS, ICONS.MYSQL],
        technologies: [
          'CPanel hosting control panel for Apache server',
          'PHP using the Wordpress framework and WooCommerce e-commerce plugin',
          'MySQL database',
          'JavaScript (jQuery, Bootstrap)',
          'Paypal Gateway API',
        ],
      },
      {
        title: 'PHP Developer',
        companyName: 'Right Left Communication',
        duration: 'Sep 2014 - Feb 2015',
        description: ['Developed, supported, monitored and maintained 4 websites for different clients from various industries.'],
        techIcons: [ICONS.PHP, ICONS.BOOTSTRAP, ICONS.HTML5, ICONS.CSS3, ICONS.JS, ICONS.JQUERY, ICONS.MYSQL],
        technologies: ['CPanel hosting control panel for Apache server', 'PHP using the Joomla framework', 'JavaScript (jQuery, Bootstrap)', 'MySQL database'],
      },
    ],
    education: [
      {
        institute: 'Navitas Professionals',
        degree: 'Professional Year Program in Computer Science',
        major: 'incl. Cert IV in Business',
        duration: 'Oct 2016 - Oct 2017',
      },
      {
        institute: 'RMIT University',
        degree: 'Bachelor of Information Technology',
        major: '',
        duration: 'Oct 2012 - Dec 2015',
      },
    ],
    projects: [
      {
        title: 'My GitHub code repository',
        description:
          'The host of my many curious projects, from setting up a custom firmware to my keyboard and my own shell configs, to testing and learning new programming languages and techniques.',
        url: 'https://github.com/samhwang',
      },
      {
        title: 'My GitHub.io page',
        description: 'This site was hosted on GitHub Pages, built with the Vite & React.',
        url: 'https://samhwang.github.io',
      },
      {
        title: 'Pi-Hole',
        description:
          'A self-hosted DNS server that will act as an ad black-hole and can be easily installed on any Raspberry Pi (hence the name, Pi-Hole), or any computer running Linux. To maintain stable and seamless connection between dropouts, I have also set up 2 of them, and sync the ad blocking lists together using a sync script.',
        url: 'https://pi-hole.net/',
      },
      {
        title: 'VAIT Discord Bot',
        description:
          'A Discord bot for the Vietnamese Aussies in IT Community, comes with built-in community requested and admin features. Built with DiscordJS and TypeScript.',
        url: 'https://github.com/viet-aus-it/vait-discord-bot',
      },
    ],
    skills: [
      {
        title: 'Professional Proficiency',
        description: [
          'Backend: PHP 7, NodeJS/TypeScript (Meteor, Express, Fastify, Serverless Framework)',
          'Frontend: jQuery, React (Redux, Gatsby, Storybook), HTML5, CSS, LESS and SASS',
          'Testing: Jest, Vitest, React Testing Framework',
          'Databases and data modelling with SQL (MariaDB/MySQL, PostgreSQL) and NoSQL (MongoDB, DynamoDB)',
          'Server configuration with Apache and NGINX',
          'Container orchestration with Docker, Docker Compose & AWS ECS',
          'Package management and build tools: Composer, NPM (AWS CodeArtifact), Yarn, Babel, Webpack, Vite, SWC, ESBuild',
          'Version control with Git, and online collaboration with GitHub, BitBucket and AWS CodeCommit',
          'Continuous Integration and Continuous Delivery with CircleCI, BuildKite, GitHub Actions',
          'Amazon Web Services products: EC2, Lambda, Elastic Beanstalk, SQS, SES, SNS, CloudWatch, S3, CloudFront, RDS',
          'Building Infrastructure as Code using AWS Cloud Development Kit (CDK)',
          'Services tracing & error monitoring tools: Sentry, Datadog',
          'Project Management tools: JIRA, Confluence, Trello, Asana',
          'Microsoft Office Suite and similar Google tools',
          'Adobe Creative Suite: Photoshop, Lightroom, Bridge',
        ],
      },
      {
        title: 'Learning',
        description: ['Golang', 'Rust', 'Deno'],
      },
    ],
    techStack: [
      ICONS.PHP,
      ICONS.COMPOSER,
      ICONS.JS,
      ICONS.TS,
      ICONS.NODE,
      ICONS.EXPRESS,
      ICONS.REACT,
      ICONS.REDUX,
      ICONS.GATSBY,
      ICONS.STORYBOOK,
      ICONS.JEST,
      ICONS.MOCHA,
      ICONS.JQUERY,
      ICONS.BABEL,
      ICONS.WEBPACK,
      ICONS.ESLINT,
      ICONS.NPM,
      ICONS.YARN,
      ICONS.HTML5,
      ICONS.CSS3,
      ICONS.LESS,
      ICONS.SASS,
      ICONS.MONGODB,
      ICONS.MYSQL,
      ICONS.POSTGRES,
      ICONS.DOCKER,
      ICONS.AWS,
      ICONS.DIGITALOCEAN,
      ICONS.GIT,
      ICONS.GITHUB,
      ICONS.GITLAB,
      ICONS.BITBUCKET,
      ICONS.TRELLO,
      ICONS.JIRA,
      ICONS.CONFLUENCE,
    ],
  },
};

export default metadata;
