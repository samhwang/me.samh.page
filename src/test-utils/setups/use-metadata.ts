import { vi } from 'vitest';
import {
  useAboutData,
  useEducationData,
  useExperienceData,
  useProjectsData,
  useSidebarData,
  useSiteMetadata,
  useSkillsData,
} from '../../../metadata/use-metadata';

vi.mock('../../../metadata/use-metadata');

const GITHUB_ICON = 'github-original';

const mockUseSiteMetadata = vi.mocked(useSiteMetadata);
mockUseSiteMetadata.mockReturnValue({
  title: 'Test Title',
  description: 'Test Description',
});

const mockUseSidebarData = vi.mocked(useSidebarData);
mockUseSidebarData.mockReturnValue({
  firstName: 'First Name',
  lastName: 'Last Name',
});

const mockUseAboutData = vi.mocked(useAboutData);
mockUseAboutData.mockReturnValue({
  firstName: 'First Name',
  lastName: 'Last Name',
  address: '123 Fake street',
  bio: 'Blah',
  socialLinks: [
    {
      icon: 'github-original',
      name: 'Github',
      url: 'https://github.com/testusername',
    },
  ],
});

const mockUseEducationData = vi.mocked(useEducationData);
mockUseEducationData.mockReturnValue([
  {
    institute: 'Bruh University',
    degree: 'Bachelor of Bruh',
    duration: 'Oct 2022 - Oct 2023',
    major: 'blah blah',
  },
]);

const mockUseExperienceData = vi.mocked(useExperienceData);
mockUseExperienceData.mockReturnValue([
  {
    title: 'Software Engineer',
    companyName: 'Bruh Industries',
    duration: 'Oct 2022 - Oct 2023',
    description: ['blah'],
    techIcons: [GITHUB_ICON],
    technologies: ['github'],
  },
]);

const mockUseProjectsData = vi.mocked(useProjectsData);
mockUseProjectsData.mockReturnValue([
  {
    title: 'Test Project',
    description: 'Test Description',
  },
]);

const mockUseSkillsData = vi.mocked(useSkillsData);
mockUseSkillsData.mockReturnValue({
  skills: [
    {
      title: 'test title',
      description: ['test desc'],
    },
  ],
  techStack: [GITHUB_ICON],
});
