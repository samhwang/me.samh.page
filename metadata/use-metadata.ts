import metadata from './metadata';

export function useSiteMetadata() {
  const { siteTitle, siteDescription } = metadata;
  return {
    title: siteTitle,
    description: siteDescription,
  };
}

export function useSidebarData() {
  const { firstName, lastName } = metadata.personalInformation;
  return {
    firstName,
    lastName,
  };
}

export function useAboutData() {
  const { firstName, lastName, bio, address, socialLinks } = metadata.personalInformation;
  return {
    firstName,
    lastName,
    bio,
    address,
    socialLinks,
  };
}

export function useEducationData() {
  const { education } = metadata.personalInformation;
  return education;
}

export function useExperienceData() {
  const { experience } = metadata.personalInformation;
  return experience;
}

interface Projects {
  title: string;
  description: string;
  url?: string;
}

export function useProjectsData(): Projects[] {
  const { projects } = metadata.personalInformation;
  return projects;
}

export function useSkillsData() {
  const { skills, techStack } = metadata.personalInformation;
  return { skills, techStack };
}
