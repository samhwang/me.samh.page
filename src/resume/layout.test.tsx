import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { useSidebarData, useSiteMetadata } from '../../metadata/use-metadata';
import RenderWrapper from '../test-utils/render-wrapper';

vi.mock('../../metadata/use-metadata');
const mockMetadataHook = vi.mocked(useSiteMetadata);
const mockSidebarHook = vi.mocked(useSidebarData);

describe('Layout rendering', () => {
  it('Should match snapshot', () => {
    const mockMetadata = {
      title: 'Test Title',
      description: 'Test Description',
    };
    mockMetadataHook.mockReturnValue(mockMetadata);
    const mockSidebarData = {
      firstName: 'First',
      lastName: 'Last',
    };
    mockSidebarHook.mockReturnValueOnce(mockSidebarData);

    const tree = render(<h1>This is a test!</h1>, { wrapper: RenderWrapper });
    expect(tree).toMatchSnapshot();
  });
});
