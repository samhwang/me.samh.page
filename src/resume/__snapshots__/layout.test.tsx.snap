// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout rendering > Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-body"
      >
        <nav
          class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"
          id="sideNav"
        >
          <a
            class="navbar-brand"
            href="#page-top"
          >
            <span
              class="d-block d-lg-none"
            >
              First Last
            </span>
            <span
              class="d-none d-lg-block"
            >
              <picture>
                <source
                  srcset="/src/assets/img/avatar.webp"
                  type="image/webp"
                />
                <source
                  srcset="/src/assets/img/avatar.jpg"
                  type="image/jpeg"
                />
                <img
                  alt="First Last Avatar"
                  class="img-fluid img-profile rounded-circle mx-auto mb-2"
                  src="/src/assets/img/avatar.jpg"
                />
              </picture>
            </span>
          </a>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse "
            id="navbarSupportedContent"
          >
            <div
              class="navbar-nav"
            >
              <ul>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="#about"
                  >
                    About
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="#experience"
                  >
                    Experience
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="#education"
                  >
                    Education
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="#skills"
                  >
                    Skills
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="#projects"
                  >
                    Projects
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
        <div
          class="container-fluid p-0"
        >
          <h1>
            This is a test!
          </h1>
        </div>
        <section
          class="_section_044294 p-3 p-lg-5 d-flex align-items-center"
        >
          <p>
            © 
            2022
             
            <a
              href="https://github.com/samhwang"
            >
              Sam Huynh
            </a>
            . Hosted on 
            <a
              href="https://app.netlify.com"
            >
              Netlify
            </a>
            . Built with 
            <a
              href="https://www.vitejs.dev/"
            >
              Vite
            </a>
             and 
            <a
              href="https://reactjs.org/"
            >
              React
            </a>
            .
          </p>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-body"
    >
      <nav
        class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"
        id="sideNav"
      >
        <a
          class="navbar-brand"
          href="#page-top"
        >
          <span
            class="d-block d-lg-none"
          >
            First Last
          </span>
          <span
            class="d-none d-lg-block"
          >
            <picture>
              <source
                srcset="/src/assets/img/avatar.webp"
                type="image/webp"
              />
              <source
                srcset="/src/assets/img/avatar.jpg"
                type="image/jpeg"
              />
              <img
                alt="First Last Avatar"
                class="img-fluid img-profile rounded-circle mx-auto mb-2"
                src="/src/assets/img/avatar.jpg"
              />
            </picture>
          </span>
        </a>
        <button
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          data-target="#navbarSupportedContent"
          data-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse "
          id="navbarSupportedContent"
        >
          <div
            class="navbar-nav"
          >
            <ul>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#about"
                >
                  About
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#experience"
                >
                  Experience
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#education"
                >
                  Education
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#skills"
                >
                  Skills
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#projects"
                >
                  Projects
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        class="container-fluid p-0"
      >
        <h1>
          This is a test!
        </h1>
      </div>
      <section
        class="_section_044294 p-3 p-lg-5 d-flex align-items-center"
      >
        <p>
          © 
          2022
           
          <a
            href="https://github.com/samhwang"
          >
            Sam Huynh
          </a>
          . Hosted on 
          <a
            href="https://app.netlify.com"
          >
            Netlify
          </a>
          . Built with 
          <a
            href="https://www.vitejs.dev/"
          >
            Vite
          </a>
           and 
          <a
            href="https://reactjs.org/"
          >
            React
          </a>
          .
        </p>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
